pkg/sql/catalog/table_elements.go

```diff
@@ -411,27 +411,85 @@ type Constraint interface {
 	// GetName returns the name of this constraint update mutation.
 	GetName() string
 
-	// NotNullColumnID returns the underlying not-null column ID, if there is one.
-	NotNullColumnID() descpb.ColumnID
-
 	// AsCheck returns the underlying check constraint, if there is one.
-	AsCheck() *descpb.TableDescriptor_CheckConstraint
+	AsCheck() CheckConstraint
 
 	// AsForeignKey returns the underlying foreign key constraint, if there is
 	// one.
-	AsForeignKey() *descpb.ForeignKeyConstraint
+	AsForeignKey() ForeignKeyConstraint
 
 	// AsUniqueWithoutIndex returns the underlying unique without index
 	// constraint, if there is one.
-	AsUniqueWithoutIndex() *descpb.UniqueWithoutIndexConstraint
+	AsUniqueWithoutIndex() UniqueWithoutIndexConstraint
 
-	// AsPrimaryKey returns the index descriptor backing the PRIMARY KEY
-	// constraint, if there is one.
-	AsPrimaryKey() Index
+	// AsUnique returns the index descriptor backing the UNIQUE or the
+	// PRIMARY KEY constraint, if there is one.
+	AsUnique() UniqueWithIndexConstraint
+}
+
+// UniqueWithIndexConstraint is an interface around a unique constraint
+// which backed by an index.
```

Comment:
nit: should the comment say "... around a UNIQUE or PRIMARY KEY constraint ..."?